static/js/components/utils/AutocompleteSettings.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import MenuItem from 'material-ui/MenuItem';
+import { DEFAULT_OPTION_LIMIT_COUNT } from '../../constants';
+
+const AutocompleteSettings = {};
+
+AutocompleteSettings.noFilter = () => true;
+
+AutocompleteSettings.defaultFilter = AutocompleteSettings.caseInsensitivePrefixFilter = (searchText, key) => {
+  let index = key.toLowerCase().indexOf(searchText.toLowerCase());
```

Comment:
This is technically outside the scope of this PR but can you replace this with `key.toLowerCase().startsWith(searchText.toLowerCase())`?